lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -165,6 +165,15 @@ static bool validateCodingKeysEnum(DerivedConformance &derived,
       propertiesAreValid = false;
       continue;
     }
+    
+    // We have a synthetized CodingKeys enum that couldn't be generated
+    // correctly because of some other error elsewhere. e.g. Invalid
+    // redeclaration of members. So let's assume this is "valid" here to
+    // skip the conformance diagnostics because the error that cause
+    // this error_type will be diagnosed elsewhere.
+    if (elt->isImplicit() && it->second->getType()->hasError()) {
```

Comment:
Looks like your case is meant to be handled below by `case TypeNotValidated`, but the path turns out to be dead due to a missing early return on `hasError()` in `varConformsToCodable`.

Reply:
> but the path turns out to be dead due to a missing early return on hasError() in varConformsToCodable.

Yeah, I thought about that ... but I wanted to avoid the `varConformsToCodable` call if we have an error type. And I think even if we return `TypeNotValidated` early for error types on  `varConformsToCodable` we still have to perform the same check again because this maybe not the only case that results in `TypeNotValidated` :)

Reply:
Why would we need to perform the same check? We ought to `continue` validating either way to diagnose other potentially extraneous keys. 

> this maybe not the only case that results in `TypeNotValidated`

`varConformsToCodable` never returns `TypeNotValidated` as of now.

Reply:
> We ought to continue validating either way to diagnose other potentially extraneous keys.

This is specific for synthesized coding keys, so besides no conformance, I can't think in an error that needs to be validated, so I think if we encounter an error here we can just return right away. 

> varConformsToCodable never returns TypeNotValidated as of now.

But as it is today the no conformance `fallthrough` to it. But that's something we can easily change :)

Reply:
> This is specific for synthesized coding keys, so besides no conformance, I can't think in an error that needs to be validated, so I think if we encounter an error here we can just return right away.

User-defined coding keys are validated here as well. It would be nice if they also benefit from this change.

```swift
struct SR_12248_1: Codable {
  enum CodingKeys: CodingKey {
    case x
  }
  var x: Int
  var x: Int
}
```

Reply:
Huum, yeah ... that makes sense I'll adjust that. Thank you :)